### PR TITLE
Add documentation describing block transforms

### PR DESCRIPTION
**Pull request by travislopes**
Here's a first pass on the documentation on defining block transforms. There's a couple transform types I haven't used before (regular expression, raw DOM node) that I could use some explanations for myself so I can add them to the docs.

### CODE REVIEW

File: docs/block-api.md [hunk 1]
```diff
@@ -109,7 +109,66 @@ attributes: {
...
+    from: [
+        {
+            type: 'block',
+            blocks: [ 'core/paragraph' ],
+            transform: ( { content } ) => {
+                return createBlock( 'core/heading', {
+                    content,
+                } );
+            },
+        },
```
Comment (youknowriad) [inline docs/block-api.md]: Should we stop here in this example and leave the shortcode for another example (to match the text above)?

File: docs/block-api.md [hunk 2]
```diff
@@ -109,7 +109,66 @@ attributes: {
...
 #### Transforms (optional)
 
-Work in progress...
+* **Type:** `Array`
+
+Transforms provide rules for what a block can be transformed from and what it can be transformed to. A block can be transformed from another block, a shortcode, a regular expression or a raw DOM node.
+
+For example, a paragraph block can be transformed into a heading block.
+
+```js
```
Comment (youknowriad) [inline docs/block-api.md]: Should we do an es5/esnext version like the other examples in the handbook?

**Comment by youknowriad** [comment]
Thanks for this, left a minor comment but this is looking good for a start. I think we probably want to move these to a separate file at some point because there's a lot to say about transforms, but it's fine for now.

**Comment by leepeterson** [comment]
👍

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Pull request merged**